File: biglist/src/test/java/org/ycode/tmaker/test/DetailDetail.java
// SPDX-License-Identifier: Apache-2.0
// (C) Panayotis Katsaloulis

package org.ycode.tmaker.test;

import onl.ycode.stormify.AutoTable;
import onl.ycode.stormify.DbField;

class DetailDetail extends AutoTable {
    @DbField(primaryKey = true)
    int id = 0;
    private String type;
    private Detail toDetail1;
    private Detail toDetail2;

    public DetailDetail(String type, Detail toDetail1, Detail toDetail2) {
        this.type = type;
        this.toDetail1 = toDetail1;
        this.toDetail2 = toDetail2;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Detail getToDetail1() {
        return toDetail1;
    }

    public void setToDetail1(Detail toDetail1) {
        this.toDetail1 = toDetail1;
    }

    public Detail getToDetail2() {
        return toDetail2;
    }

    public void setToDetail2(Detail toDetail2) {
        this.toDetail2 = toDetail2;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
}

File: db/src/commonTest/kotlin/test/DetailDetail.kt
// SPDX-License-Identifier: Apache-2.0
// (C) Panayotis Katsaloulis
package test

import onl.ycode.stormify.AutoTable
import onl.ycode.stormify.DbField

internal class DetailDetail(var type: String, toDetail1: Detail, toDetail2: Detail) : AutoTable() {
    @DbField(primaryKey = true)
    var id: Int = 0
    private var toDetail1: Detail
    private var toDetail2: Detail

    init {
        this.toDetail1 = toDetail1
        this.toDetail2 = toDetail2
    }

    fun getToDetail1(): Detail {
        return toDetail1
    }

    fun setToDetail1(toDetail1: Detail) {
        this.toDetail1 = toDetail1
    }

    fun getToDetail2(): Detail {
        return toDetail2
    }

    fun setToDetail2(toDetail2: Detail) {
        this.toDetail2 = toDetail2
    }
}
